enterprise_subsidy/apps/subsidy/models.py

```diff
@@ -380,9 +387,15 @@ def _create_redemption(
         ledger_transaction.state = TransactionStateChoices.PENDING
         ledger_transaction.save()
 
+        try:
+            if self.geag_fulfillment_handler().can_fulfill(ledger_transaction):
+                self.geag_fulfillment_handler().fulfill(ledger_transaction)
+        except Exception as exc:
+            self.rollback_transaction(ledger_transaction)
+            raise exc
+
         try:
             enterprise_fulfillment_uuid = self.enterprise_client.enroll(lms_user_id, content_key, ledger_transaction)
-            # TODO: once GEAG support is implemented we should pass external_reference (obj) to `commit_transactions`
```

Comment:
comment removed, but not fulfilled

Reply:
@pwnage101 an earlier version of our idea here was to take the geag id into the commit but, we want all committed transactions to be consistent and we may have multiple external fulfillment ids so, we opted to always commit with the enterprise enrollment. make sense?

Reply:
I'm not sure if you're aware, but Sheehan built up some code and models to allow recording an arbitrary number of "external fulfillment IDs", which this function uses under the hood.  Check out the [ExternalFulfillmentProvider](https://github.com/openedx/openedx-ledger/blob/e340ed60fca5f838b66714659afcf785a07d0f93/openedx_ledger/models.py#L347) and [ExternalTransactionReference](https://github.com/openedx/openedx-ledger/blob/e340ed60fca5f838b66714659afcf785a07d0f93/openedx_ledger/models.py#L388) models from openedx_ledger.

You may first need to create the ExternalFulfillmentProvider records via django admin, then set it in settings so that we know which provider to use from the db, then this code should be creating a new ExternalTransactionReference object and passing it to `commit_transaction(..., external_reference=external_reference)`.  At least, that's the latest plan as I know it.

Reply:
I was tempted to recommend a data migration to seed the new type (geag allocation id), but I'm recalling this comment: https://github.com/openedx/openedx-ledger/pull/15#discussion_r1153333314 which introduces the concept of a type model specifically to avoid writing code to add a new type.  Guess I'm still on the fence about that because writing code to do this seems less error-prone than manually entering it into django admin then getting the settings right so that we know which record to use.